homeassistant/components/knx/light.py

```diff
@@ -137,83 +133,77 @@ def max_mireds(self) -> int:
         return self._max_mireds
 
     @property
-    def effect_list(self) -> list[str] | None:
-        """Return the list of supported effects."""
-        return None
-
-    @property
-    def effect(self) -> str | None:
-        """Return the current effect."""
-        return None
-
-    @property
-    def is_on(self) -> bool:
-        """Return true if light is on."""
-        return bool(self._device.state)
+    def color_mode(self) -> str | None:
+        """Return the color mode of the light."""
+        if self._device.supports_rgbw:
+            return COLOR_MODE_RGBW
+        if self._device.supports_color:
+            return COLOR_MODE_RGB
+        if (
+            self._device.supports_color_temperature
+            or self._device.supports_tunable_white
+        ):
+            return COLOR_MODE_COLOR_TEMP
+        if self._device.supports_brightness:
+            return COLOR_MODE_BRIGHTNESS
+        return COLOR_MODE_ONOFF
 
     @property
-    def supported_features(self) -> int:
-        """Flag supported features."""
-        flags = 0
-        if self._device.supports_brightness:
-            flags |= SUPPORT_BRIGHTNESS
-        if self._device.supports_color:
-            flags |= SUPPORT_COLOR | SUPPORT_BRIGHTNESS
+    def supported_color_modes(self) -> set | None:
+        """Flag supported color modes."""
+        modes = set()
         if self._device.supports_rgbw:
-            flags |= SUPPORT_COLOR | SUPPORT_WHITE_VALUE
+            modes.add(COLOR_MODE_RGBW)
+        if self._device.supports_color:
+            modes.add(COLOR_MODE_RGB)
         if (
             self._device.supports_color_temperature
             or self._device.supports_tunable_white
         ):
-            flags |= SUPPORT_COLOR_TEMP
-        return flags
+            modes.add(COLOR_MODE_COLOR_TEMP)
+        if self._device.supports_brightness and not modes:
+            # Must be the only supported mode
+            modes.add(COLOR_MODE_BRIGHTNESS)
+        if not modes:
+            # Must be the only supported mode
+            modes.add(COLOR_MODE_ONOFF)
+        return modes
```

Comment:
The `color_mode` function suggests KNX will only support a single color mode.
If so, this function can be simplified to:
```py
    def supported_color_modes(self) -> set | None:
        """Flag supported color modes."""
        return {self.color_mode}
```

Reply:
The color_mode functions signature is `-> str | None` - how is it supposed to support multiple modes?

I guess there are RGBWW lights in KNX too so they could support COLOR_MODE_RGB(W) and COLOR_MODE_COLOR_TEMP. But I have to check how such are implemented.

Reply:
`self.color_mode` should reflect the current state of the light.
`self.supported_color_modes` is the set of states supported by the light.
Based on the current implementation in the PR, `self.color_mode` depends only on the light's capabilities, not on its state, and if that's the case `self.supported_color_modes` boils down to `{self.color_mode}`.